libs/service-portal/endorsements/src/screens/ViewPetition/ViewPetition.tsx

```diff
@@ -141,231 +147,254 @@ const ViewPetition = () => {
 
   return (
     <Box>
-      <Box marginBottom={5}>
-        <Text variant="h2" marginBottom={3}>
-          {petition?.title}
-        </Text>
-        <Text variant="default" marginBottom={3}>
-          {petition?.description}
-        </Text>
-
-        <Box
-          display={['block', 'flex']}
-          justifyContent="spaceBetween"
-          width={viewTypeEdit ? 'half' : 'full'}
-        >
-          {!viewTypeEdit && petition?.closedDate && (
-            <Box>
-              <Text variant="h4">{formatMessage(m.viewPetition.openTil)}</Text>
-              <Text variant="default" marginBottom={3}>
-                {format(new Date(petition?.closedDate), 'dd.MM.yyyy')}
-              </Text>
-            </Box>
-          )}
-
-          <Box>
-            <Text variant="h4">
-              {formatMessage(m.viewPetition.numberSigned)}
-            </Text>
-            <Text variant="default" marginBottom={3}>
-              {
-                (petitionEndorsements as PaginatedEndorsementResponse)
-                  .totalCount
-              }
+      {Object.entries(petition).length !== 0 ? (
```

Comment:
This has become a very deeply nested component. Would it be a good idea to take some parts and create some non-exported smaller components in the same file?

Reply:
Yes, it would! I will followup on this